app/src/main/AndroidManifest.xml

```diff
@@ -3,7 +3,11 @@
     xmlns:tools="http://schemas.android.com/tools"
     package="com.example.yard">
 
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.INTERNET" />
+
     <application
+        android:name=".ParseApplication"
```

Comment:
The PR can be split further, consider to include one module/function per PR.

Right now for your single PR, you have being doing to followings:
- Add Internet access
- Add Parse Application and dependency
- Add account creation data model
- Implement login logics
- Add post activity
- Polish singup

Having multiple features/modules to be implemented in one single PR is not recommended due to the following reasons:
- It will not lead to a clear commit title or description. For example, you current commit title or description is not reflecting all the features included in your current PR.
- Make it much harder for reviewing or getting approval since the PR is very long.
- When referring to history, the changed files are not closely related and makes it harder to debug.